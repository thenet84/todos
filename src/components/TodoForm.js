import React, { Component } from 'react';
import PropTypes from 'prop-types';

export class TodoForm extends Component {
  state = {
    title: '',
  };

  onChange = event => {
    this.setState({ [event.target.name]: event.target.value });
  };

  onSubmit = event => {
    event.preventDefault();
    this.setState({ title: '' });
    this.props.addTodo(this.state.title);
  };

  render() {
    return (
      <form onSubmit={this.onSubmit}>
        <input type="text" name="title" value={this.state.title} onChange={this.onChange} />
        <button className="button" type="submit">
          Add
        </button>
      </form>
    );
  }
}

TodoForm.propTypes = {
  addTodo: PropTypes.func.isRequired,
};
