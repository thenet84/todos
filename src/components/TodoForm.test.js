import React from 'react';
import { shallow } from 'enzyme';

import { TodoForm } from './TodoForm';

describe('TodoForm Component', () => {
  let onChangeSpy;
  let onSubmitSpy;
  let props;

  beforeEach(() => {
    onChangeSpy = jest.fn();
    onSubmitSpy = jest.fn();
    props = {
      onChange: onChangeSpy,
      onSubmit: onSubmitSpy,
      term: '',
    };
  });

  it('should render correctly', () => {
    const wrapper = shallow(<TodoForm {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should change the input test while writing', () => {
    const wrapper = shallow(<TodoForm {...props} />);

    wrapper.find('input').simulate('change', {
      target: { value: 'hello' },
    });

    expect(props.onChange).toHaveBeenCalled();
    expect(props.onChange).toHaveBeenCalledWith({
      target: { value: 'hello' },
    });
  });

  it('should submit the form', () => {
    const wrapper = shallow(<TodoForm {...props} />);

    wrapper.find('form').simulate('submit', {
      target: { value: 'hello' },
    });

    expect(props.onSubmit).toHaveBeenCalled();
    expect(props.onSubmit).toHaveBeenCalledWith({
      target: { value: 'hello' },
    });
  });
});
