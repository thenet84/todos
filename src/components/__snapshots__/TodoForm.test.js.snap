// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoForm Component should render correctly 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <input
    onChange={[MockFunction]}
    value=""
  />
  <button
    type="submit"
  >
    Submit
  </button>
</form>
`;
